packages/manager/apps/catalog-revamp/src/utils/utils.ts

```diff
@@ -0,0 +1,140 @@
+  );
+  const uniqueCategories = [
+    ...new Set(itemsFiltered.map((item) => item.category)),
+  ];
+  return uniqueCategories.sort((a, b) =>
+    a.localeCompare(b, undefined, { sensitivity: 'base' }),
+  );
+};
+
+export const matchCategories = (
+  product: Product,
+  selectedCategories: string[],
+) => {
+  return filterByEnum(product, selectedCategories, 'category');
+};
+
+export const matchUniverses = (
+  product: Product,
+  selectedUniverses: string[],
+) => {
+  return filterByEnum(product, selectedUniverses, 'universe');
+};
+
+export const matchSearchText = (product: Product, searchText: string) => {
+  const search = searchText?.toLowerCase();
+  return (
+    !search ||
+    product.category.toLowerCase().includes(search) ||
+    product.universe.toLowerCase().includes(search) ||
+    product.name.toLowerCase().includes(search)
+  );
+};
+
+export const filterProducts = (
+  products: Product[],
+  selectedCategories: string[],
+  selectedUniverses: string[],
+  searchText: string,
+) => {
+  return products.filter(
+    (product) =>
+      matchCategories(product, selectedCategories) &&
+      matchUniverses(product, selectedUniverses) &&
+      matchSearchText(product, searchText),
+  );
+};
+
+// getUniverses overloads
+export function getUniverses(
+  products: Product[],
+  withCounter: true,
+): { universe: string; count: number }[];
+export function getUniverses(
+  products: Product[],
+  withCounter?: false,
+): string[];
+// getUniverses implementation
+export function getUniverses(products: Product[], withCounter = false): any {
+  if (withCounter) {
+    const universeCounts = products.reduce((acc, product) => {
+      acc[product.universe] = (acc[product.universe] || 0) + 1;
+      return acc;
+    }, {} as Record<string, number>);
+
+    const formattedUniverses = Object.entries(
+      universeCounts,
+    ).map(([universe, count]) => ({ universe, count }));
+
+    return formattedUniverses.sort((a, b) =>
+      a.universe.localeCompare(b.universe, undefined, { sensitivity: 'base' }),
+    );
+  }
+
+  const uniqueUniverses = [
+    ...new Set(products.map((product) => product.universe)),
+  ];
+  return uniqueUniverses.sort((a, b) =>
+    a.localeCompare(b, undefined, { sensitivity: 'base' }),
+  );
+}
+
+export const getAvailableCategoriesWithCounter = (
+  products: Product[],
+  selectedUniverses: string[],
+) => {
+  const itemsFiltered = products.filter((product) =>
+    filterByEnum(product, selectedUniverses, 'universe'),
+  );
+
+  const categoryCounts = itemsFiltered.reduce((acc, product) => {
+    acc[product.category] = (acc[product.category] || 0) + 1;
+    return acc;
+  }, {} as Record<string, number>);
+
+  const formattedCategories = Object.entries(
+    categoryCounts,
+  ).map(([category, count]) => ({ category, count }));
+
+  return formattedCategories.sort((a, b) =>
+    a.category.localeCompare(b.category, undefined, { sensitivity: 'base' }),
+  );
```

Comment:
```
const countAndFormat = (objectList: any[], property: string) => {
  const countByProperty = objectList.reduce((acc, obj) => {
    acc[obj[property]] = (acc[obj[property]] || 0) + 1;
    return acc;
  }, {} as Record<string, number>);

  return Object.entries(countByProperty)
    .map(([key, count]) => ({ [property]: key, count }))
    .sort((a, b) => a[property].localeCompare(
      b[property],
      undefined,
      { sensitivity: 'base' }
     ));
};
```

```
return countAndFormat(productInSelectedUniverses, 'category');
```

or 

```
return countAndFormat(products, 'universe');
```

Reply:
Ok, on the solution above I had the typescript error `Property 'localeCompare' does not exist on type 'unknown'.`, it's solving with few more lines:
~~~
const countAndFormat = (objectList: any[], property: string) => {
  const countByProperty = objectList.reduce((acc, obj) => {
    acc[obj[property]] = (acc[obj[property]] || 0) + 1;
    return acc;
  }, {} as Record<string, number>);

  return Object.entries(countByProperty)
    .map(([key, count]) => ({ [property]: key, count }))
    .sort((a, b) => {
      // Type assertion to clarify that a[property] and b[property] are strings
      const aValue = a[property] as string;
      const bValue = b[property] as string;
      return aValue.localeCompare(bValue, undefined, { sensitivity: 'base' });
    });
};
~~~